src/devices/cpu/z180/z180asci.cpp

```diff
@@ -0,0 +1,287 @@
+		m_asci_cntla = (m_asci_cntla & Z180_CNTLA1_MPBR_EFR) | Z180_CNTLA1_CKA1D;
+		m_asci_cntlb = (m_asci_cntlb & Z180_CNTLB1_MPBT) | 0x07;
+		m_asci_stat = Z180_STAT1_TDRE;
+	}
+}
+
+uint8_t z180asci_channel::cntla_r()
+{
+	uint8_t data = m_asci_cntla;
+	LOG("Z180 CNTLA%d rd $%02x\n", m_id, data);
+	return data;
+}
+
+uint8_t z180asci_channel::cntlb_r()
+{
+	uint8_t data = m_asci_cntlb;
+	LOG("Z180 CNTLB%d rd $%02x\n", m_id, data);
+	return data;
+}
+
+uint8_t z180asci_channel::stat_r()
+{
+	uint8_t data = m_asci_stat;
+	LOG("Z180 STAT%d  rd $%02x\n", m_id, data);
+	return data;
+}
+
+uint8_t z180asci_channel::tdr_r()
+{
+	uint8_t data = m_asci_tdr;
+	LOG("Z180 TDR%d   rd $%02x\n", m_id, data);
+	return data;
+}
+
+uint8_t z180asci_channel::rdr_r()
+{
+	uint8_t data = m_asci_rdr;
+	LOG("Z180 RDR%d   rd $%02x\n", m_id, data);
+	m_asci_stat &= ~0x80;
+
+	return data;
+}
+
+void z180asci_channel::cntla_w(uint8_t data)
+{
+	LOG("Z180 CNTLA%d wr $%02x\n", m_id, data);
+	m_asci_cntla = data;
+	set_data_frame(1, (data & 4) ? 8 : 7, (data & 2) ? PARITY_ODD : PARITY_NONE, (data & 2) ? STOP_BITS_2 : STOP_BITS_1);
+}
+
+void z180asci_channel::cntlb_w(uint8_t data)
+{
+	LOG("Z180 CNTLB%d wr $%02x\n", m_id, data);
+	m_asci_cntlb = data;
+
+	int divisor = 1<<(m_asci_cntlb & 0x07);
+	divisor *= ((m_asci_cntlb & 0x20) ? 30 : 10);
+	divisor *= ((m_asci_cntlb & 0x08) ? 64 : 16);
+
+	set_tra_rate(clock(), divisor);
+	set_rcv_rate(clock(), divisor);
+}
+
+void z180asci_channel::stat_w(uint8_t data)
+{
+	if (m_id==0) {
+		LOG("Z180 STAT0  wr $%02x ($%02x)\n", data,  data & (Z180_STAT0_RIE | Z180_STAT0_TIE));
+		m_asci_stat = (m_asci_stat & ~(Z180_STAT0_RIE | Z180_STAT0_TIE)) | (data & (Z180_STAT0_RIE | Z180_STAT0_TIE));
+	} else {
+		LOG("Z180 STAT1  wr $%02x ($%02x)\n", data,  data & (Z180_STAT1_RIE | Z180_STAT1_CTS1E | Z180_STAT1_TIE));
+		m_asci_stat = (m_asci_stat & ~(Z180_STAT1_RIE | Z180_STAT1_CTS1E | Z180_STAT1_TIE)) | (data & (Z180_STAT1_RIE | Z180_STAT1_CTS1E | Z180_STAT1_TIE));
+	}
+}
+
+void z180asci_channel::tdr_w(uint8_t data)
+{
+	LOG("Z180 TDR%d   wr $%02x\n", m_id, data);
+	m_asci_tdr = data;
+	m_asci_stat &= ~0x02; // clear TDRE
+	transmit_byte(m_asci_tdr);
+}
+
+void z180asci_channel::tra_callback()
+{
+	m_tx_handler(transmit_register_get_data_bit());
+}
+
+void z180asci_channel::tra_complete()
+{
+	device_buffered_serial_interface::tra_complete();
+	m_asci_stat |= 0x02; // set TDRE
+	if (m_asci_stat & Z180_STAT0_TIE) {
+		m_irq = 1;
+	}
+}
+
+void z180asci_channel::rdr_w(uint8_t data)
+{
+	LOG("Z180 RDR%d   wr $%02x\n", m_id, data);
+	m_asci_rdr = data;
+}
+
+void z180asci_channel::received_byte(u8 byte)
+{
+	m_asci_stat |= 0x80;
+	m_asci_rdr = byte;
+	if (m_asci_stat & Z180_STAT0_RIE) {
+		m_irq = 1;
+	}
+}
+
+DECLARE_WRITE_LINE_MEMBER( z180asci_channel::write_rx )
+{
+	device_buffered_serial_interface::rx_w(state);
+}
+
+int z180asci_channel::check_interrupt()
+{
+	return m_irq;
+}
+void z180asci_channel::clear_interrupt()
+{
+	m_irq = 0;
+}
```

Comment:
These could probably do with being inlined.

Reply:
The compiler should be smart enough to inline it on its own, and there's the potential that the functions could grow in complexity. This doesn't add technical debt nor does it fail to match the rest of the codebase. I agree with most of the feedback, but this one seems to be going overboard.

Reply:
In this case it can’t though, because the calls come from z180.cpp, so it can’t see the implementation inside z180asci.cpp to realise it’s simple enough to inline (unless you turn on link-time code generation which we don’t because it’s a lot of overhead for little if any gain in most cases).